src/presenter/event-presenter.js

```diff
@@ -1,35 +1,90 @@
-  tripComponent = new TripView();
   sortComponent = new SortView();
-  eventListComponet = new EventListView();
-  editComponent = new EventEditView();
+  eventListComponent = new EventsListView();
 
-  constructor ({mainContainer}){
-    this.mainContainer = mainContainer;
+  constructor({ container, eventsModel }) {
+    this.container = container;
+    this.eventsModel = eventsModel;
   }
 
-  init() {
-    render(this.mainComponent, this.mainContainer);
-    render(this.tripComponent, this.mainComponent.getElement());
+  renderEventItem(callback) {
+    const itemComponent = new EventItemView();
+    render(itemComponent, this.eventListComponent.getElement());
+    callback(itemComponent);
+  }
 
-    const tripEventElement = this.tripComponent.getElement();
-    render(this.sortComponent, tripEventElement);
+  renderEventNew() {
+    this.renderEventItem((container) => {
+      const event = {
+        id: null,
+        basePrice: 0,
+        dateFrom: null,
+        dateTo: null,
+        destination: null,
+        isFavorite: null,
+        offers: [],
+        type: 'flight',
+      };
+      const offers = this.eventsModel.getOffersByType(event.type);
+      render(
+        new EventEditView({
+          event: event,
+          cities: this.cities,
+          offers: offers,
+        }),
+        container.getElement()
+      );
+    });
+  }
+
+  renderEventEdit(event) {
+    this.renderEventItem((container) => {
+      const city = this.eventsModel.getCityById(event.destination);
+      const offers = this.eventsModel.getOffersByType(event.type);
+      render(
+        new EventEditView({
+          event: event,
+          city: city,
+          cities: this.cities,
+          offers: offers,
+        }),
+        container.getElement()
+      );
+    });
+  }
 
-    render(this.eventListComponet, tripEventElement);
-    for (let i = 0; i < 4; i++) {
-      const eventItemComponent = new EventItemView();
-      render(eventItemComponent, this.eventListComponet.getElement());
-      render((i === 0) ? this.editComponent : new EventView(), eventItemComponent.getElement());
+  renderTripPoints() {
+    for (let i = 0; i < this.events.length; i++) {
+      this.renderEventItem((container) => {
+        const event = this.events[i];
+        const city = this.eventsModel.getCityById(event.destination);
+        const offers = this.eventsModel.getSelectedOffers(
+          event.type,
+          event.offers
+        );
+        render(
+          new EventView({ event: event, city: city, offers: offers }),
+          container.getElement()
+        );
+      });
     }
   }
+
+  init() {
+    this.events = [...this.eventsModel.getEvents()];
+    this.cities = [...this.eventsModel.getCities()];
+    this.offers = [...this.eventsModel.getOffers()];
+
+    render(this.sortComponent, this.container);
+    render(this.eventListComponent, this.container);
+    this.renderEventEdit(this.events[0]);
+    this.renderTripPoints();
```

Comment:
Проверить по критериям и ТЗ количество элементов.

```suggestion
    this.renderEventEdit(this.events[0]);
    this.renderTripPoints();
```